mob/src/main/java/com/github/phantazmnetwork/mob/spawner/BasicMobSpawner.java

```
@@ -3,64 +3,169 @@
+            if (!Modifier.isPublic(method.getModifiers()) || method.getReturnType() != void.class) {
+                continue;
+            }
+
+            Parameter[] parameters = method.getParameters();
+            if (parameters.length != 1) {
+                continue;
+            }
+
+            String methodName = method.getName();
+            if (!methodName.startsWith("set")) {
+                continue;
+            }
+            String key;
+            if (methodName.length() > 3) {
+                key = Character.toLowerCase(methodName.charAt(3)) + methodName.substring(4);
+            }
+            else {
+                key = methodName.substring(3);
+            }
+
+            ConfigElement element = metaNode.getElementOrDefault((ConfigElement)null, key);
+            if (element == null) {
+                continue;
+            }
+
+            Parameter parameter = parameters[0];
+            NotNull[] notNulls = parameter.getAnnotationsByType(NotNull.class);
+            Nullable[] nullables = parameter.getAnnotationsByType(Nullable.class);
+            Class<?> type = parameter.getType();
+            boolean optional = !type.isPrimitive() && nullables.length != 0 || notNulls.length < 1;
+
+            ConfigProcessor<?> processor = processorMap.get(BooleanObjectPair.of(optional, type.getName()));
+            if (processor == null) {
+                continue;
+            }
+
+            Object data;
+            try {
+                data = processor.dataFromElement(element);
+            }
+            catch (ConfigProcessException e) {
+                e.printStackTrace();
+                continue;
+            }
+            if (data instanceof Optional<?> dataOptional) {
+                data = dataOptional.orElse(null);
+            }
+            try {
+                method.invoke(meta, data);
+            }
+            catch (IllegalAccessException | InvocationTargetException e) {
+                e.printStackTrace();
```

another potential case for logging, alternatively, rethrowing the exception, don't really see a reason to swallow it here?